File: flutter/packages/workern_models/lib/notification.dart
/// Notification data model for Workern apps (matches TypeScript WorkernNotification)
class WorkernNotification {
  final String id;
  final String title;
  final String description;
  final String type;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool seen;
  final String? imageUrl;
  final Map<String, dynamic>? data;

  WorkernNotification({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.createdAt,
    required this.updatedAt,
    this.seen = false,
    this.imageUrl,
    this.data,
  });

  /// Create from Firestore document data
  factory WorkernNotification.fromFirestore(Map<String, dynamic> json) {
    return WorkernNotification(
      id: json['id'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      type: json['type'] ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'].toDate().toString())
          : DateTime.now(),
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'].toDate().toString())
          : DateTime.now(),
      seen: json['seen'] ?? false,
      imageUrl: json['imageUrl'],
      data: json['data'],
    );
  }

  /// Convert to Firestore document format
  Map<String, dynamic> toFirestore() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'type': type,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
      'seen': seen,
      'imageUrl': imageUrl,
      'data': data,
    };
  }
}

File: tools/firebase/functions-ts/src/models/notification.ts
import { Timestamp } from 'firebase-admin/firestore';
export class Notification {
  id: string;
  title: string;
  description: string;
  type: string;
  createdAt: Timestamp;
  seen = false;
  data: any;

  constructor(params: any = {}) {
    this.id = params.id;
    this.title = params.title || '';
    this.description = params.description || '';
    this.type = params.type || '';
    this.seen = params.seen || false;
    this.createdAt = params.createdAt || Timestamp.now();
    this.data = params.data || null;
  }

  forFirestore() {
    const notification = Object.assign({}, this);
    return notification;
  }
}
